File: jobfolio-backend/src/main/java/com/harshadcodes/jobfolio/entity/Application.java
package com.harshadcodes.jobfolio.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "applications")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Application {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;

    private String jobTitle;
    private String company;
    private String jobUrl;
    private String location;
    private String source;

    @Enumerated(EnumType.STRING)
    private ApplicationStatus status = ApplicationStatus.APPLIED;

    private String lastStatusSource = "MANUAL";

    private LocalDateTime appliedDate = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();
}

File: frontend/src/types/application.ts
export type ApplicationStatus =
  | 'APPLIED' | 'RECEIVED' | 'ASSESSMENT' | 'INTERVIEW'
  | 'OFFER' | 'ACCEPTED' | 'REJECTED';

export interface Application {
  id: number;
  jobTitle: string;
  company: string;
  jobUrl: string | null;
  location: string | null;
  source: string | null;
  status: ApplicationStatus;
  lastStatusSource: string | null; // 'MANUAL' | 'EMAIL_SYNC' | null
  appliedDate: string;
  updatedAt: string;
}

export interface ApplicationEvent {
  oldStatus: ApplicationStatus | null;
  newStatus: ApplicationStatus;
  source: 'MANUAL' | 'EMAIL_SYNC';
  createdAt: string;
}

export interface CreateApplicationPayload {
  jobTitle: string;
  company: string;
  jobUrl?: string;
  location?: string;
  source?: string;
}